common/src/allocations/monitor.rs

```diff
@@ -67,18 +69,29 @@ impl AllocationMonitor {
         network_subgraph: &NetworkSubgraph,
         graph_network_id: u64,
     ) -> Result<u64> {
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct GraphNetwork {
+            current_epoch: u64,
+        }
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct QueryResult {
+            graph_network: GraphNetwork,
+        }
+
         let res = network_subgraph
-            .network_query(
-                r#"
+            .query::<QueryResult>(&json!({
+                "query": r#"
                     query epoch($id: ID!) {
                         graphNetwork(id: $id) {
                             currentEpoch
                         }
                     }
-                "#
-                .to_string(),
-                Some(serde_json::json!({ "id": graph_network_id })),
-            )
+                "#,
+                "variables":
+            { "id": graph_network_id },
```

Comment:
formatting here is a bit off
```suggestion
                "variables": { "id": graph_network_id },
```